pkg/virt-controller/services/launchercommandrenderer.go

```diff
@@ -0,0 +1,155 @@
+package services
+
+import (
+	"strconv"
+
+	"github.com/google/go-cmp/cmp"
+	"kubevirt.io/client-go/log"
+)
+
+type VirtLauncherCommandRenderer struct {
+	allowEmulation           bool
+	customDebugFilters       string
+	domainName               string
+	isDoppelgangerPod        bool
+	isNonRoot                bool
+	gracePeriodSeconds       int
+	numberOfHookSidecars     int
+	keepAfterFailure         bool
+	namespace                string
+	shouldSimulateCrash      bool
+	virtLauncherStaticConfig VirtLauncherStaticConfig
+	vmUID                    string
+}
+
+type VirtLauncherStaticConfig struct {
+	containerDiskDir string
+	ephemeralDiskDir string
+	launcherTimeout  int
+	ovmfPath         string
+	virtShareDir     string
+}
+
+type VirtLauncherCommandRendererOption func(renderer *VirtLauncherCommandRenderer)
+
+func NewVirtLauncherCommandRenderer(
+	vmUID string,
+	domainName string,
+	gracePeriodSeconds int,
+	hookSidecarListLen int,
+	vmNamespace string,
+	virtLauncherStaticConfig VirtLauncherStaticConfig,
+	opts ...VirtLauncherCommandRendererOption,
+) *VirtLauncherCommandRenderer {
+
+	return applyOptionsToRenderer(
+		&VirtLauncherCommandRenderer{
+			domainName:               domainName,
+			gracePeriodSeconds:       gracePeriodSeconds,
+			numberOfHookSidecars:     hookSidecarListLen,
+			namespace:                vmNamespace,
+			virtLauncherStaticConfig: virtLauncherStaticConfig,
+			vmUID:                    vmUID,
+		},
+		opts,
+	)
+}
+
+func applyOptionsToRenderer(virtLauncherCmdRenderer *VirtLauncherCommandRenderer, opts []VirtLauncherCommandRendererOption) *VirtLauncherCommandRenderer {
+	for _, opt := range opts {
+		if opt == nil {
+			continue
+		}
+		opt(virtLauncherCmdRenderer)
+	}
+	return virtLauncherCmdRenderer
+}
+
+func NewDoppelgangerPodRender(opts ...VirtLauncherCommandRendererOption) *VirtLauncherCommandRenderer {
+	return applyOptionsToRenderer(
+		&VirtLauncherCommandRenderer{isDoppelgangerPod: true},
+		opts,
+	)
+}
+
+func (vlcr *VirtLauncherCommandRenderer) Render() []string {
+	if vlcr.isDoppelgangerPod {
+		if vlcr.doesDoppelgangerPodHaveOptions() {
+			log.Log.Warning("doppelganger pod does not allow options")
+		}
+		return []string{"/bin/bash",
+			"-c",
+			"echo", "bound PVCs"}
+	}
+
+	const launcherExecBinary = "/usr/bin/virt-launcher-monitor"
+	command := []string{
+		launcherExecBinary,
+		"--qemu-timeout", generateQemuTimeoutWithJitter(vlcr.virtLauncherStaticConfig.launcherTimeout),
```

Comment:
I know this is mostly moved code, but wouldn't it make sense to have only the binary in `command` and then use `args` for the ... args?